crates/node/build.rs

```diff
@@ -1,6 +1,18 @@
+use std::env;
+
 use substrate_build_script_utils::{generate_cargo_keys, rerun_if_git_head_changed};
 
 fn main() {
+    // Check if the feature flag is enabled
+    let feature_enabled = env::var("CARGO_FEATURE_DEV").is_ok();
+    // Check if we are in release mode
+    let release_mode = env::var("PROFILE").map(|p| p == "release").unwrap_or(false);
```

Comment:
we should also check for production mode which is the one actually used no?

Reply:
ahh good catch, changed it so that it works on debug mode only